tests/testsuite/publish.rs

```diff
@@ -132,14 +132,18 @@ You may press ctrl-c to skip waiting; the crate should be available shortly.
 
 #[cargo_test]
 fn duplicate_version() {
+    let arc: Arc<Mutex<u32>> = Arc::new(Mutex::new(0));
     let registry_dupl = RegistryBuilder::new()
         .http_api()
         .http_index()
-        .add_responder("/api/v1/crates/new", move |_req, _server| Response {
-            code: 200,
-            headers: vec![],
-            body: br#"{"errors": [{"detail": "crate version `0.0.1` is already uploaded"}]}"#
-                .to_vec(),
+        .add_responder("/index/3/f/foo", move |req, server| {
+            let mut lock = arc.lock().unwrap();
+            *lock += 1;
+            if *lock <= 1 {
+                server.not_found(req)
+            } else {
+                server.index(req)
+            }
         })
```

Comment:
If we have a published package, would `RegistryBuilder` do this for us or does it not correctly handle these cases?

Reply:
We do need a custom responder before the fix because our dev registry does not report duplicate package errors, so we can't just double publish to hit the test case.